src/tribler/core/utilities/async_group/async_group.py

```diff
@@ -3,38 +3,38 @@
 from contextlib import suppress
 from typing import Iterable, List, Set
 
+from tribler.core.utilities.async_group.exceptions import CancelledException
+
 
 class AsyncGroup:
     """This class is a little brother of TaskManager and his purpose is only to
     correct cancel a group asyncio Tasks/Futures
 
     Example:
-    >>> from tribler.core.utilities.async_group import AsyncGroup
+    >>> from tribler.core.utilities.async_group.async_group import AsyncGroup
     >>> async def void():
     ...     pass
     >>> group = AsyncGroup()
-    >>> group.add([
-    ...     void(),
-    ...     void(),
-    ...     void()
-    ... ])
+    >>> group.add_task(void())
+    >>> group.add_task(void())
+    >>> group.add_task(void())
     >>> await group.cancel()
     """
 
     def __init__(self):
         self._futures: Set[Future] = set()
+        self._cancelled = False
 
-    def add(self, *coroutines) -> List[Task]:
+    def add_task(self, coroutine) -> Task:
```

Comment:
It is possible to add an argument type here
```suggestion
    def add_task(self, coroutine: Coroutine) -> Task:
```